src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsClient/DiagnosticsClient.cs

```diff
@@ -380,6 +381,47 @@ internal async Task DisablePerfMapAsync(CancellationToken token)
             ValidateResponseMessage(response, nameof(DisablePerfMapAsync));
         }
 
+        /// <summary>
+        /// Create a new DiagnosticsClient instance using the specified diagnostic port.
+        /// </summary>
+        /// <param name="diagnosticPort">The diagnostic port.</param>
+        /// <param name="ct">The token to monitor for cancellation requests.</param>
+        public static async Task<DiagnosticsClient> FromDiagnosticPort(string diagnosticPort, CancellationToken ct)
+        {
+            if (diagnosticPort is null)
+            {
+                throw new ArgumentNullException(nameof(diagnosticPort));
+            }
+
+            IpcEndpointConfig portConfig = IpcEndpointConfig.Parse(diagnosticPort);
+
+            if (portConfig.IsListenConfig)
+            {
+                string fullPort = RuntimeInformation.IsOSPlatform(OSPlatform.Windows) ? portConfig.Address : Path.GetFullPath(portConfig.Address);
+                ReversedDiagnosticsServer server = new(fullPort);
+                server.Start();
+
+                try
+                {
+                    IpcEndpointInfo endpointInfo = await server.AcceptAsync(ct).ConfigureAwait(false);
+                    return new DiagnosticsClient(endpointInfo.Endpoint);
+                }
+                catch (TaskCanceledException)
+                {
+                    //clean up the server
+                    await server.DisposeAsync().ConfigureAwait(false);
```

Comment:
This is likely problematic as the endpoint that is passed to the DiagnosticsClient has a reference to the ReversedDiagnosticsServer. Invoking nearly any method on the client is going to throw an ObjectDisposedException.

I would recommend validating the current behavior and adding tests for this method before releasing this API addition.

cc @dotnet/dotnet-diag

Reply:
> This is likely problematic as the endpoint that is passed to the DiagnosticsClient has a reference to the ReversedDiagnosticsServer. Invoking nearly any method on the client is going to throw an ObjectDisposedException.

Hm, this code was mostly taken from [here](https://github.com/dotnet/diagnostics/blob/7b73d2d6c5e3691efc9b4b5608ff1125aa8e6666/src/Tools/Common/ReversedServerHelpers/ReversedServerHelpers.cs#L245-L267)

Though I'm skeptical that it is going to generate `ObjectDisposedException` in that case. That being said there is definitely a missing dispose on the server once the DiagnosticClient is disposed.

I can make a PR to cover the dispose of the DiagnosticClient to also dispose the server. If that's ok?

Reply:
But DiagnosticClient doesn't have currently a dispose so... so I would have to add it. That would not break existing code but new code using the diagnostic port would have to call dispose.

Reply:
Proposing #5073 as a fix, let me know

Reply:
> Though I'm skeptical that it is going to generate `ObjectDisposedException` in that case. That being said there is definitely a missing dispose on the server once the DiagnosticClient is disposed.

You are correct. I mistakenly read the catch as a finally. And your observation that the server is not disposed is correct as well.

I still think it's problematic (now for a different reason) because the server is not disposed which will case the Unix domain socket file to not be cleaned up or cause the socket file (on Unix-like machines) or the named pipe (on Windows machines) to dangle and not be resuable during the lifetime of the process or until a garbage collection kicks in, if the client ever gets garbage collected.